mcbackend/core.py

```diff
@@ -181,16 +182,20 @@ def to_inferencedata(self, **kwargs) -> InferenceData:
         chain_lengths = {c.cid: len(c) for c in chains}
         if len(set(chain_lengths.values())) != 1:
             _log.warning("Chains vary in length. Lenghts are: %s", chain_lengths)
-
+        
+        if equalize_chain_lengths:
+            # A minimum chain length is introduced so that all chains have equal length
+            clen = min(chain_lengths.values())
```

Comment:
```suggestion
        clen = None
        if equalize_chain_lengths:
            # A minimum chain length is introduced so that all chains have equal length
            clen = min(chain_lengths.values())
```